### ISSUE TITLE
linter: .gitignore not respected by default

### ISSUE THREAD

**Issue by webpro** [oxc-project/oxc#11901]
### What version of Oxlint are you using?

1.3.0

### What command did you run?

oxlint

### What happened?

The local `.gitignore` file is not respected by default.

```sh
oxlint --ignore-path=.gitignore
```

This works fine, but maybe consider having this as the default behavior? I think it's expected behavior for more users than current.

If the behavior is updated, then for edge cases maybe there should be an `--no-ignore`.

**Comment by nrayburn-tech** [comment]
The .gitignore file should already be respected. Any chance you have a reproduction?

The code looks like it only uses the .gitignore file from the current working directory, maybe that's a problem.

**Comment by nrayburn-tech** [comment]
It seems like even nested .gitignore files are respected. I used https://github.com/oxc-project/oxc-intellij-plugin/tree/main/sandbox/nested-configs for testing.

- Without any changes, there are 7 warnings and 8 errors.
- Adding `package-c` to https://github.com/oxc-project/oxc-intellij-plugin/blob/main/sandbox/nested-configs/.gitignore and I receive 5 warnings and 7 errors.
- Creating a .gitignore within https://github.com/oxc-project/oxc-intellij-plugin/tree/main/sandbox/nested-configs/package-c and adding `index.js` to it and I receive 5 warnings and 7 errors.

**Comment by webpro** [comment]
In an empty directory:

```
touch index.js
mkdir dist
touch dist/index.js
pnpm dlx oxlint
# Finished in 1ms on 2 files with 87 rules using 10 threads.
pnpm dlx oxlint --version
# Version: 1.3.0
echo "dist" > .gitignore
pnpm dlx oxlint
# Finished in 1ms on 2 files with 87 rules using 10 threads.
```

But now I see the issue here: there hasn't been a `git init`. The other day I was copying around template files and the target was not a Git repo yet.

Sorry, invalid.
